### PR TITLE
FIRAppAttestArtifactStorage implementation and tests

### PR DESCRIPTION
**Pull request by maksymmalyhin**
#no-changelog

### CODE REVIEW

File: FirebaseAppCheck/Tests/Unit/AppAttestProvider/Storage/FIRAppAttestArtifactStorageTests.m [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  self.appName = @"FIRAppAttestArtifactStorageTests";
+  self.appID = @"1:100000000000:ios:aaaaaaaaaaaaaaaaaaaaaaaa";
+
+  self.storage = [[FIRAppAttestArtifactStorage alloc] initWithAppName:self.appName
+                                                                appID:self.appID
+                                                          accessGroup:nil];
+}
+
+- (void)tearDown {
+  self.storage = nil;
```
Comment (ncooke3) [inline FirebaseAppCheck/Tests/Unit/AppAttestProvider/Storage/FIRAppAttestArtifactStorageTests.m:45]: Maybe call `[self.storage setArtifact:nil]` before `self.storage = nil` to clean up left over storage from `testSetAndGetArtifact` (so the left over storage doesnt persist between test runs).

File: FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestArtifactStorage.m [hunk 2]
```diff
@@ -22,16 +22,82 @@
...
+
+- (FBLPromise<NSData *> *)getArtifact {
+  return [self.keychainStorage getObjectForKey:[self artifactKey]
+                                   objectClass:[NSData class]
+                                   accessGroup:self.accessGroup]
+      .then(^NSData *(id<NSSecureCoding> storedArtifact) {
+        if ([(NSObject *)storedArtifact isKindOfClass:[NSData class]]) {
+          return (NSData *)storedArtifact;
+        } else {
+          return nil;
```
Comment (ncooke3) [inline FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestArtifactStorage.m:75]: Should a promised rejected with some error be returned instead?

**Review by ncooke3 — APPROVED** [review]
LGTM after addressing comments!

File: FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestArtifactStorage.h [hunk 3]
```diff
@@ -25,9 +25,14 @@ NS_ASSUME_NONNULL_BEGIN
 @protocol FIRAppAttestArtifactStorageProtocol <NSObject>
 
 /// Set the artifact.
+/// @param artifact The artifact data to store.
+/// @return An artifact that is resolved with the artifact data passed into the method in case of
+/// success or is rejected with an error.
```
Comment (ncooke3) [inline FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestArtifactStorage.h:30]: optional: mention that passing `nil` will clear any existing artifact data from storage.

**Review by ncooke3 — APPROVED** [review]

File: FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestArtifactStorage.m (hunk 2 above)
Comment (maksymmalyhin) [reply FirebaseAppCheck/Sources/AppAttestProvider/Storage/FIRAppAttestArtifactStorage.m:75]: Maybe... The method should return `nil` if the artifact was not found. The case when something else was stored in the keychain looks more similar to "not found" case so we eventually should overwrite the value. WDYT?

**Pull request merged**